File: NicolasManrique_GenerarTiempos.py
import numpy as np
import time


def fibonacci(N):
    if N==0:
        return 0
    elif N==1:
        return 1
    else:
        return fibonacci(N-1)+fibonacci(N-2)


def get_time(N):
    t0=time.time()
    fibonacci(N)
    t1=time.time()-t0
    return t1

archivo=open("times_python.csv","w")

for i in range(35):    
    archivo.write(str(i))
    archivo.write(",")
    archivo.write(str(get_time(i)))
    archivo.write("\n")
    

File: NicolasManrique_GenerarTiempos.cpp
#include <iostream>
#include <math.h>
#include <ctime>

using namespace std;


int fibonacci (int N);

float get_time(int N);

int main () {
    for(int i = 0; i<35;i+=1){
       cout << i;
       cout << ",";
       cout << get_time(i); 
       cout << "\n";
   }
   return 0;
}



int fibonacci(int N) {

   if(N==0){
      return 0;
    }  
   else if(N==1){
      return 1;
    }
   else{
      return fibonacci(N-1)+fibonacci(N-2);

   }
}


float get_time(int N){

    clock_t t;
    t=clock();
    fibonacci(N);
    t=clock()-t;
    float time = ((float)t)/CLOCKS_PER_SEC;
    return time;

}


